Project/Gem/Source/Navigation/SplinePosesPublisher.cpp

```diff
@@ -0,0 +1,141 @@
+
+namespace ROS2::Demo
+{
+
+    void SplinePosesPublisher::Reflect(AZ::ReflectContext* context)
+    {
+        if (AZ::SerializeContext* serialize = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serialize->Class<SplinePosesPublisher>()
+                ->Version(1)
+
+                ->Field("ReverseDirection", &SplinePosesPublisher::m_reverseDirection)
+                ->Field("PathTopicConfig", &SplinePosesPublisher::m_topicName)
+                ->Field("PoseCount", &SplinePosesPublisher::m_poseCount);
+
+            if (AZ::EditContext* ec = serialize->GetEditContext())
+            {
+                ec->Class<SplinePosesPublisher>("SplinePosesPublisher", "SplinePosesPublisher")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                    ->Attribute(AZ::Edit::Attributes::Category, "ROS2")
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC_CE("Game"))
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &SplinePosesPublisher::m_reverseDirection,
+                        "Reverse Direction",
+                        "Reverse the robot movement direction")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default, &SplinePosesPublisher::m_topicName, "Topic Name", "Topic name to publish poses")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &SplinePosesPublisher::m_poseCount,
+                        "Pose Count",
+                        "Number of poses to create and publish");
+            }
+        }
+    }
+
+    void SplinePosesPublisher::GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required)
+    {
+        required.push_back(AZ_CRC_CE("SplineService"));
+    }
+
+    void SplinePosesPublisher::Activate()
+    {
+        auto ros2Node = ROS2Interface::Get()->GetNode();
+
+        CalculatePoses();
+
+        rclcpp::QoS qos(5);
+        qos.reliability(rclcpp::ReliabilityPolicy::Reliable).durability(rclcpp::DurabilityPolicy::TransientLocal);
+
+        m_pathPublisher = ros2Node->create_publisher<nav_msgs::msg::Path>(m_topicName.data(), qos);
+        m_pathPublisher->publish(m_path);
+    }
+
+    void SplinePosesPublisher::Deactivate()
+    {
+    }
+
+    void SplinePosesPublisher::CalculatePoses()
+    {
+        AZ::ConstSplinePtr splinePtr{ nullptr };
+        LmbrCentral::SplineComponentRequestBus::EventResult(splinePtr, m_entity->GetId(), &LmbrCentral::SplineComponentRequests::GetSpline);
+        AZ_Assert(splinePtr, "Spline pointer is null");
+
+        AZ::Transform splineTransform{ AZ::Transform::CreateIdentity() };
+        AZ::TransformBus::EventResult(splineTransform, m_entity->GetId(), &AZ::TransformBus::Events::GetWorldTM);
+
+        const size_t steps = m_poseCount - 1;
+
+        m_path.poses.reserve(m_poseCount);
```

Comment:
m_path does not have its header initialized. This should be initialized so that the paths can be visualized in rviz.